internal/context/prompts.go

```diff
@@ -0,0 +1,129 @@
+package context
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"regexp"
+
+	"github.com/commitdev/zero/internal/config/moduleconfig"
+	"github.com/commitdev/zero/pkg/util/exit"
+	"github.com/manifoldco/promptui"
+)
+
+type PromptHandler struct {
+	moduleconfig.Parameter
+	Condition func(map[string]string) bool
+}
+
+func NoCondition(map[string]string) bool {
+	return true
+}
+func KeyMatchCondition(key string, value string) func(map[string]string) bool {
+	return func(param map[string]string) bool {
+		return param[key] == value
+	}
+}
+
+// TODO: validation / allow prompt retry ...etc
+func (p PromptHandler) GetParam(projectParams map[string]string) string {
```

Comment:
Seems like this would be a good one to have a test for, exercising the different types.

Reply:
i did, i aliased my `git a` to `git add . -u` and forgot to index the untracked file 😅